src/async/mutationHooks/user/useUpdateUserProfileOptimisticMutation.ts

```diff
@@ -13,30 +13,36 @@ export default function useUpdateUserProfileOptimisticMutation() {
 	const updateUserProfileOptimisticMutation = useMutation({
 		mutationFn: updateUserProfile,
 		onMutate: async (data) => {
-			// Cancel any outgoing refetches (so they don't overwrite our optimistic update)
+			// First, cancel any outgoing refetches
 			await queryClient.cancelQueries({
 				queryKey: [USER_PROFILE, session?.user?.id],
 			});
 
 			// Snapshot the previous value
 			const previousData = queryClient.getQueryData([
-				CATEGORIES_KEY,
+				USER_PROFILE,
```

Comment:
@rogerantony-dev  What's the reason for changing this from `CATEGORIES_KEY` to `USER_PROFILE`